Revise developer documentation

t-b opened the pull request:
Remove CONTRIBUTE.md as it is superfluous.

Rework INSTALL.md with the follwoing goals:
- Describe all CMake variables in a easy-to-read manner
- Explain how cross-compiling tango works
- Explain how to compile the tango dependencies

@bourtemb @mliszcz Sorry for rushing this now.

But I would need that for tuesday. I've also given you write access to the forked repo. So for bigger changes please push them as fixup commits. This is way more convenient than github comments.

t-b committed:
Revise developer documentation

Rework INSTALL.md with the follwoing goals:
- Describe all CMake variables in a easy-to-read manner
- Explain how to compile the tango dependencies

mliszcz commented on INSTALL.md:
```diff
@@ -1,96 +1,190 @@
+mkdir build
+cd build
+cmake ..
+make [-j NUMBER_OF_CPUS]
```
You don't need this step, 'all' target is empty in tango-idl, and anyway '-j' won't speed up anything in such simple package.

BTW I noticed that you can't run these steps without c++ compiler being installed, which makes no sense. But maybe this requirement comes from CMake.

t-b replied:
Good point. Done.

mliszcz commented on configure/toolchain-i686.cmake:
```diff
@@ -0,0 +1,15 @@
+set(CMAKE_CROSSCOMPILING TRUE)
```
IMO this file should be merged as part of #792 or at least as a separate commit in this PR, just to keep the history clear.

t-b replied:
@mliszcz Yeah #792 would be a good fit as well, but as I wanted it to be ready for tomorrow I snuck it in here. I've now extracted it into its own commit. Thanks for reminding me.

mliszcz reviewed (dismissed):
Looks good to me, thanks for cleaning this file up!

bourtemb reviewed (commented):
The old CONTRIBUTE.md was still in Draft mode since it was created and not updated since then.
I think it could be useful to keep a CONTRIBUTE.md file because it integrates well with Github interface.
See https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/setting-guidelines-for-repository-contributors
Maybe we should update CONTRIBUTE.md and simply put a link to the wiki page as in The README.md file?

t-b commented:
@bourtemb The wiki page is pretty ugly as well. But I can link to it from CONTRIBUTE.md if you think this is useful.

bourtemb commented:
> @bourtemb The wiki page is pretty ugly as well. But I can link to it from CONTRIBUTE.md if you think this is useful.

I think it is useful if the wiki page is up to date. Please feel free to update it if you think it can be improved.
I think the original idea behind creating the wiki page was to make it easy to write and update.
So please feel free to improve it.

t-b commented:
@bourtemb Okay I'll have a look.

t-b committed:
CONTRIBUTING.md: Polish and make it the only place

We had some contributing related information in README.md as well but
CONTRIBUTING.md is a better fit.

The old file CONTRIBUTE.md was a mere draft version and also named wrong
according to [1].

[1]: https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/setting-guidelines-for-repository-contributors

t-b committed:
README.md: Cleanup

Remove duplicated newlines and link section.

t-b committed:
Add cross-compilation toolchain file and instructions

This allows users to compile for 32bit on a 64bit linux. This serves
mostly as a starting point for more fancy cross compilation projects and
for integrating that in our own CI.

t-b commented:
@bourtemb Done. I've moved the wiki stuff into CONTRIBUTING.md as that is the github suggested way of doing things. And I also like it better thant he wiki :)

bourtemb commented on README.md:
```diff
@@ -55,36 +54,10 @@ API reference: [tango-controls.github.io/cppTango-docs](https://tango-controls.g
-# Links
-
-[Tango Controls web page](http://tango-controls.org)
+See [CONTRIBUTE file](INSTALL.md).
```
```suggestion
See [CONTRIBUTING file](CONTRIBUTING.md).
```

bourtemb committed:
Fix link to CONTRIBUTING file in README.md

t-b commented:
@bourtemb @mliszcz Thanks both.
